models/_sector/nft/trades/solana/platforms/tensorswap_solana_trades.sql

```diff
@@ -0,0 +1,260 @@
+            , t.current_price as amount_raw
+            , t.current_price/1e9 as amount_original
+            , t.current_price/1e9 * sol_p.price as amount_usd
+            , 'SOL' as currency_symbol
+            , 'So11111111111111111111111111111111111111112' as currency_address
+            , t.account_metadata
+            , tk.account_master_edition
+            , tk.account_mint
+            , tk.verified_creator
+            , 'TSWAPaqyCSx2KABk68Shruf4rp7CxcNi8hAsbdwmHbN' as project_program_id
+            , cast(null as varchar) as aggregator_name
+            , cast(null as varchar) as aggregator_address
+            , t.call_tx_id as tx_id
+            , t.call_block_slot as block_slot
+            , t.call_tx_signer as tx_signer
+            --taker fees + maker fees = platform fees. Tensorswap implemented a 0.4% maker fee rebate on 8/22, that is paid out of taker fees.
+            , t.tswap_fee as taker_fee_amount_raw
+            , t.tswap_fee/1e9 as taker_fee_amount
+            , t.tswap_fee/1e9 * sol_p.price as taker_fee_amount_usd
+            , case when t.tswap_fee = 0 then 0 else t.tswap_fee/t.current_price end as taker_fee_percentage
+            , -1*(case when t.call_block_time > timestamp '2023-05-25' then 0.004*t.current_price else 0 end) as maker_fee_amount_raw
+            , -1*(case when t.call_block_time > timestamp '2023-05-25' then 0.004*t.current_price else 0 end)/1e9 as maker_fee_amount
+            , -1*(case when t.call_block_time > timestamp '2023-05-25' then 0.004*t.current_price else 0 end)/1e9 * sol_p.price as maker_fee_amount_usd
+            , -1*(case when t.call_block_time > timestamp '2023-05-25' then 0.004 else 0 end) as maker_fee_percentage
+            --amm fees
+            , t.mm_fee as amm_fee_amount_raw
+            , t.mm_fee/1e9 as amm_fee_amount
+            , t.mm_fee/1e9 * sol_p.price as amm_fee_amount_usd
+            , case when t.mm_fee = 0 then 0 else t.mm_fee/t.current_price end as amm_fee_percentage
+            , t.creators_fee as royalty_fee_amount_raw
+            , t.creators_fee/1e9 as royalty_fee_amount
+            , t.creators_fee/1e9 * sol_p.price as royalty_fee_amount_usd
+            , case when t.creators_fee = 0 then 0 else t.creators_fee/t.current_price end as royalty_fee_percentage
+            , t.instruction
+            , t.outer_instruction_index
+            , coalesce(t.inner_instruction_index, 0) as inner_instruction_index
+        FROM trades t
+        LEFT JOIN {{ ref('tokens_solana_nft') }} tk
+            ON t.account_metadata = tk.account_metadata
+        LEFT JOIN {{ source('prices', 'usd') }} sol_p ON sol_p.blockchain = 'solana' and sol_p.symbol = 'SOL' and sol_p.minute = date_trunc('minute', t.call_block_time) --get sol_price
+    )
+
+SELECT
+*
+, concat(project,'-',trade_category,'-',cast(outer_instruction_index as varchar),'-',cast(coalesce(inner_instruction_index,0) as varchar),'-',account_metadata,'-',tx_id) as unique_trade_id
```

Comment:
Do we need this for UX purposes or is it added due to habit (we used to have this as the `unique_key` above)? Otherwise we should drop this since these tend to take up quite a lot of space. 

If you think it's makes the table nicer to query though I am not against it. We could also use `{{ dbt_utils.generate_surrogate_key(['field_a', 'field_b'[,...]]) }}` so that it becomes a hash which is smaller.

Reply:
oh meant to drop this